src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntity.scala

```diff
@@ -25,4 +26,15 @@ case class AgoraEntity(@(ApiModelProperty@field)(required = false, value = "The
                        payload: Option[String] = None,
                        @(ApiModelProperty@field)(required = false, value = "URI for method details")
                        url: Option[String] = None
-                        )
\ No newline at end of file
+                        )
+
+object AgoraProjectionDefaults {
+  val RequiredProjectionFields = Seq[String]("namespace", "name", "id")
+}
+
+case class AgoraEntityProjection(includedFields: Seq[String], excludedFields: Seq[String]) {
+  require(excludedFields.intersect(RequiredProjectionFields).isEmpty)
+  require(includedFields.isEmpty || excludedFields.isEmpty)
+
```

Comment:
Yeah - sorry was reading that the wrong way.  Forget the second line I added to the note.